### PR TITLE
Make slack channel optional

### PR DESCRIPTION
**Pull request by SomtochiAma**
## Changes
- Makes slack channel optional
- Only redacts token from logs if it is present (to prevent distorting the error message)
- Checks status code of HTTP request and returns an error if not successful

Fixes: #200 

Signed-off-by: Somtochi Onyekwere <somtochionyekwere@gmail.com>

### CODE REVIEW

File: internal/notifier/client.go [hunk 1]
```diff
@@ -87,9 +88,18 @@ func postMessage(address, proxy string, certPool *x509.CertPool, payload interfa
 	for _, o := range reqOpts {
 		o(req)
 	}
-	if _, err := httpClient.Do(req); err != nil {
+	resp, err := httpClient.Do(req)
+	if err != nil {
 		return fmt.Errorf("failed to execute request: %w", err)
 	}
 
+	if resp.StatusCode != http.StatusOK {
```
Comment (stefanprodan) [inline internal/notifier/client.go:96]: What about HTTP 202?

Comment (SomtochiAma) [reply internal/notifier/client.go:96]: Will add that in

File: internal/notifier/client.go [hunk 2]
```diff
@@ -87,9 +88,18 @@ func postMessage(address, proxy string, certPool *x509.CertPool, payload interfa
...
+	if err != nil {
 		return fmt.Errorf("failed to execute request: %w", err)
 	}
 
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusAccepted && resp.StatusCode != http.StatusCreated {
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("unable to read response body, %s", err)
+		}
+		return fmt.Errorf("request not successful, %s", string(b))
```
Comment (stefanprodan) [inline internal/notifier/client.go:101]:
```suggestion
		return fmt.Errorf("request failed with status code %: %s", resp.StatusCode, string(b))
```

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @SomtochiAma 🌷

**Pull request merged**